### ISSUE TITLE
Configuration of SharePoint for Viva Learning - cannot be saved

### ISSUE THREAD

**Issue by DCasl** [MicrosoftDocs/Viva#640]
In Teams, in Viva Learning, under the Admin tab, I have tried to add a SharePoint provider. Long before that, on the company's communications site I created a library meant for storing learning materials. I have the GlobalAdmin rights, and I am one of the owners of the communication site.
According to the instruction I followed the steps to configure a SharePoint provider and got the message "Your changes couldn't be saved. Make sure you've entered the valid data."
![Web capture_14-7-2023_15321_teams microsoft com](https://github.com/MicrosoftDocs/Viva/assets/121768994/1c600771-df61-4f44-85c8-42ca3de83151)

What have I done wrong to configure SharePoint in Viva Learning and what should I fix?

**Comment by scanum** [comment]
Thank you for submitting feedback.
I think the best way forward is if you open a service ticket in your tenant so this can get resolved ASAP. Based on the outcome let me know if it is something that can be called out in the docs.
Please follow this link to contact support for business products:https://docs.microsoft.com/office365/admin/contact-support-for-business-products
Please keep us posted here on the resolution so we can feed whatever information you discover into the content.

**Comment by DCasl** [comment]
I will open a service ticket. Thank you very much!

**Comment by scanum** [comment]
Thank you @DCasl!

**Comment by EricDeferm01** [comment]
Hi @DCasl did you manage to find out what the issue was (we're experiencing similar issue).? Tx

**Comment by DCasl** [comment]
Hi @EricDeferm01,
I opened a service ticket and shortly after that was phone called, then emailed by the support team. The instructions included running the Fidler trace app https://docs.microsoft.com/office365/troubleshoot/support/run-fiddler-trace. I repeated the same steps to configure the SharePoint provider several times with the Fidler trace running (by mistake I ran the Microsoft Support and Recovery Assistant app, not the one for Microsoft 365). 
When I was repeating those steps, in one of those attempts I succeeded and got the "Success" message. I checked the repository SharePoint list, and it was finally created. I waited for the folders with learning material to show up in VivaLearning and the files were ingested in VivaLearning, but not all. Logs showed a lot of failures, mentioning that "Every item needs to have Microsoft 365 group permission to light-up inside Viva Learning", though the settings for each library in which folders were located included permissions for the team associated with the communication site where the repository was created. Solved that later by changing group permissions.
Anyway, I just tried the same actions repeatedly to add the SharePoint provider and the only variation was using the URL for the site
"https://_domain_.sharepoint.com/_LibraryName_/Forms/AllItems.aspx", and later "https://_domain_.sharepoint.com/". But I have tried this before and it did not work. 
Hope this helps!

**Comment by EricDeferm01** [comment]
Hi @DCasl , I managed to define SharePoint as resource for Viva Learning. In our case the issue was caused by selecting one of the new predefined templates when creating a SP Communication site. After creating a standard Communication site (without selecting a template) the changes to save the SharePoint configuration were applied w/ succes.
![image](https://github.com/MicrosoftDocs/Viva/assets/47383152/0f1cb1be-e5ba-4a1b-bc2c-cd56d224f11d)
